[BUG] get_sdist_metadata return value in tests can change

```
(py38) denolf@lindenolf:~/dev/grayskull$ pip list | grep souschef
conda-souschef                2.2.1
(py38) denolf@lindenolf:~/dev/grayskull$ pip list | grep gray
grayskull                     1.5.1.dev3+g62509c4.d20220728      (current main branch)
```

This test passes when I run it like this

```
pytest tests/test_pypi.py::test_compose_test_section_with_console_scripts
```

but fails when I do this

```
pytest tests/test_pypi.py
```

Somehow `get_sdist_metadata` returns something different in both cases. Not sure why.

Some debugging led me to `injection_distutils`.

It seems this method is absorbing any exception (comment: never absorb `BaseException`, but that's another discussion)

```python
@contextmanager
def injection_distutils(folder: str, debug=False) -> dict:
     ...
    try:
        __run_setup_py(path_setup, data_dist)
        ...
    except BaseException as err:
        raise
        log.debug(f"Exception occurred when executing sdist injection: {err}")
        yield data_dist
    finally:
        core.setup = setup_core_original
        os.chdir(old_dir)
```

The exception we get when running `pytest tests/test_pypi.py` is this

```python
_____________________________________________________________________________________ test_compose_test_section_with_console_scripts _____________________________________________________________________________________

    def test_compose_test_section_with_console_scripts():
        config = Configuration(name="pytest", version="7.1.2")
        metadata1 = get_pypi_metadata(config)
>       metadata2 = get_sdist_metadata(
            "https://pypi.io/packages/source/p/pytest/pytest-7.1.2.tar.gz", config, debug=True
        )

tests/test_pypi.py:247: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
grayskull/strategy/py_base.py:694: in get_sdist_metadata
    with injection_distutils(temp_folder, debug=debug) as metadata:
/users/denolf/.pyenv/ubuntu_20_04/versions/3.8.5/lib/python3.8/contextlib.py:113: in __enter__
    return next(self.gen)
grayskull/strategy/py_base.py:385: in injection_distutils
    __run_setup_py(path_setup, data_dist)
grayskull/strategy/py_base.py:442: in __run_setup_py
    merge_sdist_metadata(data_dist, get_setup_cfg(os.path.dirname(str(path_setup))))
grayskull/strategy/py_base.py:289: in get_setup_cfg
    setup_cfg = read_configuration(str(path_setup_cfg))
/users/denolf/virtualenvs/ewoks/ubuntu_20_04/py38/lib/python3.8/site-packages/setuptools/config.py:87: in read_configuration
    dist = Distribution()
/users/denolf/virtualenvs/ewoks/ubuntu_20_04/py38/lib/python3.8/site-packages/setuptools/dist.py:459: in __init__
    _Distribution.__init__(
/users/denolf/.pyenv/ubuntu_20_04/versions/3.8.5/lib/python3.8/distutils/dist.py:292: in __init__
    self.finalize_options()
/users/denolf/virtualenvs/ewoks/ubuntu_20_04/py38/lib/python3.8/site-packages/setuptools/dist.py:837: in finalize_options
    ep(self)
/tmp/grayskull-pytest-d060832v/pytest-5.3.1/.eggs/setuptools_scm-7.0.5-py3.8.egg/setuptools_scm/integration.py:123: in infer_version
    _assign_version(dist, config)
/tmp/grayskull-pytest-d060832v/pytest-5.3.1/.eggs/setuptools_scm-7.0.5-py3.8.egg/setuptools_scm/integration.py:59: in _assign_version
    maybe_version = _get_version(config)
/tmp/grayskull-pytest-d060832v/pytest-5.3.1/.eggs/setuptools_scm-7.0.5-py3.8.egg/setuptools_scm/__init__.py:154: in _get_version
    parsed_version = _do_parse(config)
/tmp/grayskull-pytest-d060832v/pytest-5.3.1/.eggs/setuptools_scm-7.0.5-py3.8.egg/setuptools_scm/__init__.py:101: in _do_parse
    version = _version_from_entrypoints(config) or _version_from_entrypoints(
/tmp/grayskull-pytest-d060832v/pytest-5.3.1/.eggs/setuptools_scm-7.0.5-py3.8.egg/setuptools_scm/_entrypoints.py:65: in _version_from_entrypoints
    for ep in iter_matching_entrypoints(root, entrypoint, config):
/tmp/grayskull-pytest-d060832v/pytest-5.3.1/.eggs/setuptools_scm-7.0.5-py3.8.egg/setuptools_scm/discover.py:65: in iter_matching_entrypoints
    for ep in iter_entry_points(entrypoint):
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

group = 'setuptools_scm.parse_scm', name = None

    def iter_entry_points(
        group: str, name: str | None = None
    ) -> Iterator[_t.EntrypointProtocol]:
        all_eps = entry_points()
        if hasattr(all_eps, "select"):
            eps = all_eps.select(group=group)
        else:
>           eps = all_eps[group]
E           KeyError: 'setuptools_scm.parse_scm'

/tmp/grayskull-pytest-d060832v/pytest-5.3.1/.eggs/setuptools_scm-7.0.5-py3.8.egg/setuptools_scm/_entrypoints.py:96: KeyError
```

The method `injection_distutils` is quite horrible. There must be a better way. In addition I've seen lots of dodgy stuff concerning `setuptools_scm` all over the code base.

--- a/setup.py
+++ b/setup.py
@@ -54,6 +54,7 @@ setup(
             "pytest-cov",
             "pytest-console-scripts",
             "pytest-mock",
+            "setuptools-scm",
         ]
     },
     url="https://github.com/conda-incubator/grayskull",

